### ISSUE TITLE
Ironhide, Nvidia and Xen kernel on Oneric 

### ISSUE THREAD

**Issue by simoneaster** [MrMEEE/ironhide#132]
I'm trying to get to the following setup
Lenovo T520 with 
Oneric (unity removed......horrible)  with  linux-image-server and xen suite installed.
nvidia manual binary driver installation with ignore xen presence set.

http://nathanosullivan.com/blog/computing/using-ubuntu-904-as-a-xen-dom0-host-with-nvidia-driver/

IGNORE_XEN_PRESENCE=y CC="gcc -DNV_VMAP_4_PRESENT -DNV_SIGNAL_STRUCT_RLIM" ./NVIDIA-Linux-x86_64-###.##-pkg2.run

The ideal is to have a laptop that provides xen/sandpit environments whilst still being useful for HD/3D stuff on the move(i.e. cloud in my bag + hotel entertainer)

Now I can get Ironhide installed but it pulls in the Oneric bundled drivers for nvidia and works, but as soon as I boot into a xen kernel this breaks because nvidia driver is not loaded due to lack of xen support.

Hopefully I can make a manual ironhide installation after binary Nvidia install?

Thanks

Simon

**Comment by MrMEEE** [comment]
I haven't tried with xen, but should definitely be possible..

I think the easiest would be for you to download the source for the ironhide deb package, and recompile it without the dependency for nvidia-current.. If you have no clue howto do this, please tell me, and i'll see what I can do...

**Comment by simoneaster** [comment]
Yes please no clue when it comes to debs :) I could do with some help on non dependency deb build.

As it is have to reinstall maybe this time with proper ubuntu-server install as opposed to desktop, reverting back to linux-image-desktop still leaves me with a 3.0.0 generic kernel and whilst glxgears runs 
optirun glxgears
optirun32 wine HomeWorld2.exe 
gives me 
###############
ERROR: ld.so: object 'libdlfaker.so' from LD_PRELOAD cannot be preloaded: ignored.
ERROR: ld.so: object 'librrfaker.so' from LD_PRELOAD cannot be preloaded: ignored.
Xlib:  extension "GLX" missing on display ":0.0".
###############

To revert  
I've cleaned out after a  gdm stop.

apt-get purge nvidia-common nvidia-current nvidia settings ironhide ironhide-ui
and 
apt-get install nvidia-common nvidia-current nvidia settings ironhide ironhide-ui

I then perform sudo ironhide-configure with CRT as manual display using the Lenovo template and see glxgears, but still the '32bit' problem.

Like I said I'll likely reinstall with xen kernel, manual nvidia binary installation then hopefully ironhide without dependencies and I'll have the dream portable lab!

Many thanks again for your work, redbull  in transit :)

**Comment by MrMEEE** [comment]
Ok...

I think I have a solution for you: 

Remove the ironhide package... 
Install the xen kernel and the driver
install the "ironhide-xen" package, which doesn't depend on the nvidia-current package..

With the problem you have on 32-bit apps, just install "virtualgl-libs:i386"..

Let me knows how it goes..

**Comment by bencolombini** [comment]
Installing virtualgl-libs:i386 got Wine apps (32 bit) working for me!

Thank you so much for your efforts, MrMEEE, without you I'd have to try to return a Windows-loaded-laptop with the excuse "It doesn't run 3d games on Linux".

**Comment by simoneaster** [comment]
Ok finally got round to the rebuild.
Manual nvidia install has gone in and installed ironhide-xen and ironhide-ui package.

KMS: build Completed.

acpi_call:
Running module version sanity check.
- Original module
  - No original module exists within this kernel
- Installation
  - Installing to /lib/modules/3.0.0-12-server/updates/dkms/

depmod....

DKMS: install Completed.
Setting up curl (7.21.6-3ubuntu3) ...
Setting up dialog (1.1-20110707-1) ...
Setting up virtualgl (2.2.99.1-1~oneiricubuntu4) ...
Setting up easybashgui (1.3.1-1~oneiricubuntu4) ...
Setting up ironhide-xen (1.0-1~oneiricubuntu1) ...
/var/lib/dpkg/info/ironhide-xen.postinst: 33: /usr/bin/ironhide-configuration: not found
Processing triggers for libc-bin ...
ldconfig deferred processing now taking place

I can't find optirun or optirun32 either.

Any ideas?

Thanks

Simon

**Comment by simoneaster** [comment]
Sounds like the deb is a bit broken, any guides how to build the src into deb without dependancies?

Thanks

Simon
